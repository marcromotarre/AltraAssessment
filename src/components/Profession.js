import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export const ProfessionContainer = styled.section`
  padding: 4px 8px;
  background: #7b7b7b;
  border-radius: 10px;
  height: 20px;
  label {
    font-family: 'HelveticaNeue-Thin';
    color: #ffffff;
    font-size: 13px;
  }
`;

const Profession = ({ name }) => {
  return (
    <ProfessionContainer>
      <label>{name}</label>
    </ProfessionContainer>
  );
};

Profession.propTypes = {
  name: PropTypes.string.isRequired,
};
export default Profession;
