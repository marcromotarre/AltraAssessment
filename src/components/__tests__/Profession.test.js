import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Profession from '../Profession';

/** this component should:
 *  - render profession:
 *      - profession name
 */

const props = {
  name: 'profession name',
};

describe('Profession Component tests', () => {
  it('Profession Snapshot', () => {
    const tree = renderer.create(<Profession {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Profession renders profession name', () => {
    const wrapper = shallow(<Profession {...props} />);

    expect(wrapper.children().length).toBe(1);
    expect(wrapper.find('label').length).toBe(1);
    const label = wrapper.childAt(0);
    expect(label.text()).toBe(`${props.name}`);
  });
});
