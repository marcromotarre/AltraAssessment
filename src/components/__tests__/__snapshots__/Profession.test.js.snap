// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profession Component tests Profession Snapshot 1`] = `
<section
  className="sc-AxjAm hHoNSr"
>
  <label>
    profession name
  </label>
</section>
`;
